crates/scheduler/src/lib.rs

```diff
@@ -60,4 +62,7 @@ pub trait BaseScheduler {
     ///
     /// `current` is the current running task.
     fn task_tick(&mut self, current: &Self::SchedItem) -> bool;
+
+    /// set priority for a task
+    fn set_priority(&mut self, task: &Self::SchedItem, _prio: isize) -> isize;
```

Comment:
Returns `bool`?

Reply:
In https://linux.die.net/man/2/setpriority: 

#### Return Value
Since getpriority() can legitimately return the value -1, it is necessary to clear the external variable errno prior to the call, then check it afterward to determine if -1 is an error or a legitimate value. **The setpriority() call returns 0 if there is no error, or -1 if there is.**

Reply:
Returns `true` for success, `false` for failure. `-1` is confusable.